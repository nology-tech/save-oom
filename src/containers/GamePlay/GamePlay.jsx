import React, { useEffect } from 'react'
import GameEnd from "../../pages/GameEnd/GameEnd"
import { useState } from 'react';


const GamePlay = () => {
  const [showComponent, setShowComponent] = useState(false);


  useEffect(() => {
    setInterval(()=> {
      setShowComponent(!showComponent);
    }, 60000)
  }, []);

  return (

    <section
     
    className='game-play'>
      <p>GameisPlaying</p>
      {showComponent && <GameEnd/>}
      

    </section>

  )
}

export default GamePlay;