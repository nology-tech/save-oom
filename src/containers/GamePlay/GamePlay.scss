@use "../../styles/variables" as *;

.game-play {



  height: 100%;
  width: 100%;

  background-color: $color-secondary-dark;
  //background-image: url("");

}
