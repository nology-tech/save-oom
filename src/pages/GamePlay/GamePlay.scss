@use '../../styles/variables' as *;
@use '../../../node_modules/animate.css/animate.css';

.game-play {
  display: flex;
  flex-direction: row;

  &__images {
    height: 100vh;
  }

  &__background {
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-size: cover;
    height: 100vh;
    width: 85vw;
  }

  &__oom {
    position: absolute;
    top: 65vh;
    left: 65vw;
    animation-delay: 1s;
    animation-duration: 10s;
  }

  &__squirrel {
    position: absolute;
    left: 20vw;
    top: 75vh;
    transform: scaleX(-1);
    animation: bounce;
    animation-duration: 10s;
    animation-delay: 2s;
  }

  &__button {
    position: absolute;
    top: 50vh;
    left: 50vw;
  }
}
