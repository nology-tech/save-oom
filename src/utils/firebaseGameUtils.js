import { db } from "../firebase";
import {
  addDoc,
  collection,
  getDocs,
  query,
  where,
  orderBy,
  Timestamp,
} from "firebase/firestore";
/** 
 * Get all the rounds of a specific game and level for a given user, ordered by most recent first 
 @param {*} userId
  @param {String} game
*@returns a querySnap (document snapshot), returns a Promise.
*/

export const getGameRoundsForUser = async (id, game) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** 
   * Get all the correct round of a specific game and level for a given user, ordered by most recent first 
  @param {String} userId
  @param {String} gameId
  @param {Number} level  
  *@returns a promise with the query snapshot.
  */
export const getCorrectGameRoundsForUser = async (id, game, level) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    where("level", "==", level),
    where("correct", "==", true),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** Get all the incorrect rounds for a given user, ordered by most recent first 
  @param {String} userId
  *@returns  a promise with the query snapshot.
  */
export const getIncorrectGameRoundsForUser = async (id) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("correct", "==", false),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** Get all the incorrect round of a specific game for a given user, ordered by most recent first 
  @param {String} userId
  @param {String} gameId
  @param {Number} level  
  *@returns  a promise with the query snapshot.
  */

export const getAllIncorrectGameRoundsForUser = async (id, game) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    where("correct", "==", true),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};



/** Function that returns an array of rounds 
  @param {String} userId
  @param {String} gameId
  @param {Function} getGameRounds
  *@returns returs a promise.
  */
export const getArrayOfRounds = async (id, game, level, getGameRounds) => {
  const querySnap = await getGameRounds(id, game, level);
  const phonicsArr = querySnap.docs.map((doc) => {
    return doc.data().phonic;
  });
  return phonicsArr;
};

/** Function that returns an array of rounds 
  @param {String} id
  @param {String} game
  @param {Object} roundInfo
  @param {String} answer
  *@returns returs a promise.
  */

export const saveUserRound = async (id, game, level, gameStats, answer) => {
  const { isCorrect, score } = gameStats;
  await addDoc(collection(db, "users", id, "rounds_played"), {
    correct: isCorrect,
    gameId: game,
    phonic: answer,
    level: level,
    roundTime: Timestamp.fromDate(new Date()).toDate(),
    score: score,
  });
};
