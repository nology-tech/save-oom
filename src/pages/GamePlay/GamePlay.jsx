import React from 'react';
import background from '../../assets/images/Frame 5backgroung.png';
import swingingOom from '../../assets/images/Group 146swingingOom.png';
import squirrel from '../../assets/images/squirrel.png';
import AnimatedImage from '../../components/AnimatedImage/AnimatedImage';
import './GamePlay.scss';

const GamePlay = () => {
  return (
    <div className='game-play'>
      <div className='game-play__images'>
        <img
          className='game-play__background'
          src={background}
          alt='background'
        />

        <AnimatedImage
          imageToAnimate={squirrel}
          animationClass={'animate__animated'}
          animationType={'animate__bounce'}
          imageStylesClass={'game-play__squirrel'}
        />
        <AnimatedImage
          imageToAnimate={swingingOom}
          animationClass={'animate__animated.animate__fast'}
          animationType={'animate__swing'}
          imageStylesClass={'game-play__oom'}
        />
      </div>
    </div>
  );
};

export default GamePlay;
