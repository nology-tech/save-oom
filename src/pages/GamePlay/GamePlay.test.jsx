import { render, screen } from '@testing-library/react';
import { customRender } from '../../utils/testUtils';
import GamePlay from './GamePlay';

it('Should render the Game Play screen', () => {
  const { container } = customRender(<GamePlay />);

  expect(container).toMatchSnapshot();
});
