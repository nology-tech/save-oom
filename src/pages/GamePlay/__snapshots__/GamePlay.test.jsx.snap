// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the AnimatedImage component on screen 1`] = `<AnimatedImage />`;

exports[`Should render the Game Play screen 1`] = `
<div>
  <div
    class="game-play"
  >
    <div
      class="game-play__images"
    >
      <img
        alt="background"
        class="game-play__background"
        src="Frame 5backgroung.png"
      />
      <img
        alt=""
        class="animate__animated animate__bounce game-play__squirrel "
        id="squirrel"
        src="squirrel.png"
      />
      <img
        alt=""
        class="animate__animated.animate__fast animate__swing game-play__oom "
        id="oom"
        src="Group 146swingingOom.png"
      />
    </div>
  </div>
</div>
`;
